File: favouriteflix-backend/src/main/java/in/favouriteflix/service/FavouriteFlixService.java
package in.favouriteflix.service;

import java.util.List;

import in.favouriteflix.entities.FavouriteFlix;

public interface FavouriteFlixService {
	public List<FavouriteFlix> getAllFavFlix();
	public FavouriteFlix addFavFlix(FavouriteFlix favouriteFlix);
	public void deleteFavFlix(String imdbID);
}

File: favouriteflix-frontend/src/services/FavouriteFlixService.js
import axios from "axios";

const api = axios.create({
  baseURL: process.env.REACT_APP_API_BASE_URL + "/api/favflix",
});

api.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem("token");

    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
    }

    return config;
  },
  (error) => {
    return Promise.reject(error);
  },
);

class FavouriteFlixService {
  addFavFlix(favFlix) {
    return api.post("/addFavFlix", favFlix);
  }

  getAllFavFlix() {
    return api.get("/getAllFavFlix");
  }

  deleteFavFlix(imdbID) {
    return api.delete("/deleteFavFlix/" + imdbID);
  }
}

export default new FavouriteFlixService();
